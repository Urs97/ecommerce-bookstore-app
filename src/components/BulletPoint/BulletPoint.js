import React from 'react';
import './BulletPoint.scss';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

const BulletPoint = ({mode, icon, title, text, color}) => {
    if(mode === "default") {
        return (
            <div className="bulletpoint">
                <span className={color}>
                    <FontAwesomeIcon icon={icon} />
                </span>
                <p>{text}</p>
            </div>
        )
    }

    if(mode === "expanded") {
        return (
            <div className="bulletpoint bulletpoint-expanded">
                <span className={color}>
                    <FontAwesomeIcon icon={icon} />
                </span>
                <h4 className="title">{title}</h4>
                <p>{text}</p>
            </div>
        )
    }

    return <div>Bulletpoint error - invalid mode prop</div>;
};

export default BulletPoint;