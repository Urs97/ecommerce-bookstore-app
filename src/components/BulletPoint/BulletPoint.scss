@import '../../styles/shared.scss';

.bulletpoint {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;

    span {
        text-align: center;
        display: flex;
        align-items: center;
        justify-content: center;
        font-size: 2.75em;
        background-color: $blue;
        width: 2.5em;
        height: 2.5em;
        border: 0.075em solid $blue;
        border-radius: 50%;

        &:hover {
            background-color: #FFFFFF;
        }
    }

    p {
        font-size: 1.25em;
        margin-top: 1em;
        text-align: center;
        color: $text-color;
        flex-grow: 1;
    }
}

.bulletpoint-expanded {

    padding: 0 2em;

    & span {
        color: $green;
        border-color: $green;
        background-color: #FFFFFF;

        &:hover {
            background-color: $green;
            color: #FFFFFF;
            border-color: $green;
        }
    }

    & h4 {
        margin-top: 1em;
        font-size: 1.5em;
        text-align: center;
    }

    & p {
        font-size: 1.15em;
    }
}