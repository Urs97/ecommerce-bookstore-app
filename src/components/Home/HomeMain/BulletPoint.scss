@import '../../../styles/shared.scss';

.bulletpoint {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;

    span {
        text-align: center;
        display: flex;
        align-items: center;
        justify-content: center;
        font-size: 2.75em;
        background-color: $blue;
        width: 2.5em;
        height: 2.5em;
        border: 0.075em solid $blue;
        border-radius: 50%;

        &:hover {
            background-color: #FFFFFF;
        }
    }

    p {
        font-size: 1.25em;
        margin-top: 1em;
        text-align: center;
        color: $text-color;
    }
}