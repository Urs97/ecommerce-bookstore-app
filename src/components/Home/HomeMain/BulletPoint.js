import React from 'react';
import './BulletPoint.scss';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

function BulletPoint(props) {
    return (
    <div className="bulletpoint">
        <span><FontAwesomeIcon icon={props.icon} /></span>
        <p>{props.text}</p>
    </div>
    )
};

export default BulletPoint;