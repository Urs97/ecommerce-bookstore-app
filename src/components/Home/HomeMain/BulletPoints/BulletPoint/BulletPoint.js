import React from 'react';
import './BulletPoint.scss';

function BulletPoint() {
    return (
        <div>BulletPoint brah</div>
    )
};

export default BulletPoint;