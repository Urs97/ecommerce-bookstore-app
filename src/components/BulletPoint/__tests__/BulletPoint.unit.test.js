import React from 'react';
import {shallow} from 'enzyme';
import BulletPoint from '../BulletPoint';

describe('BulletPoint', () => {
    test('renders empty BulletPoint', () => {
        const wrapper = shallow(
            <BulletPoint />
        );
        expect(wrapper).toMatchSnapshot();
    });
});