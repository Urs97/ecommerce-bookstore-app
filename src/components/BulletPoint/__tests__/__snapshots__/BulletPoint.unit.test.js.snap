// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BulletPoint renders BulletPoint in expanded mode 1`] = `
<div
  className="bulletpoint bulletpoint-expanded"
>
  <span
    className="yellow"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon="mockIcon.svg"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
  </span>
  <h4
    className="title"
  >
    BulletPoint title
  </h4>
  <p>
    This is BulletPoint text
  </p>
</div>
`;

exports[`BulletPoint renders BulletPoint in normal mode 1`] = `
<div>
  Bulletpoint error - invalid mode prop
</div>
`;

exports[`BulletPoint renders BulletPoint in unrecognizable mode 1`] = `
<div>
  Bulletpoint error - invalid mode prop
</div>
`;
